org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/decorators/params/byname/CallByNameParamAtCreationPresenter.scala

```diff
@@ -0,0 +1,96 @@
+package org.scalaide.ui.internal.editor.decorators.params.byname
+
+import org.scalaide.core.extensions.SemanticHighlightingParticipant
+import org.scalaide.core.internal.jdt.model.ScalaCompilationUnit
+import scala.reflect.internal.util.SourceFile
+import org.scalaide.core.compiler.IScalaPresentationCompiler
+import org.eclipse.jface.text.Position
+import org.eclipse.jface.text.source.Annotation
+import org.scalaide.logging.HasLogger
+import org.eclipse.jface.text.source.ISourceViewer
+import org.scalaide.ui.internal.editor.decorators.BaseSemanticAction
+import org.scalaide.core.internal.compiler.ScalaPresentationCompiler
+import org.scalaide.ui.internal.preferences.CallByNameParamCreationPreferencePage
+import org.scalaide.core.IScalaPlugin
+
+class CallByNameParamAtCreationPresenter(sourceViewer: ISourceViewer) extends
+  BaseSemanticAction(sourceViewer, CallByNameParamAtCreationAnnotation.ID, Some("callByNameParamCreation")) {
+
+  import CallByNameParamAtCreationPresenter._
+
+  protected override def findAll(compiler: ScalaPresentationCompiler, scu: ScalaCompilationUnit, sourceFile: SourceFile): Map[Annotation, Position] =
+    CallByNameParamAtCreationPresenter.annotations(compiler, scu, sourceFile, prefStoreCfg)
+}
+
+object CallByNameParamAtCreationPresenter extends HasLogger {
+  case class Cfg(firstLineOnly: Boolean)
```

Comment:
A case class with one parameter and no super trait seems unnecessary, returning the boolean directly should be sufficient.